server/utils/courseParticipationUtils.ts

```diff
@@ -3,13 +3,81 @@ import type { Request } from 'express'
 import DateUtils from './dateUtils'
 import StringUtils from './stringUtils'
 import { referPaths } from '../paths'
-import type { CourseParticipation, CourseParticipationWithName, Referral } from '@accredited-programmes/models'
+import type {
+  CourseParticipation,
+  CourseParticipationOutcome,
+  CourseParticipationSetting,
+  CourseParticipationUpdate,
+  CourseParticipationWithName,
+  Referral,
+} from '@accredited-programmes/models'
 import type {
   GovukFrontendSummaryListRowWithValue,
   GovukFrontendSummaryListWithRowsWithValues,
 } from '@accredited-programmes/ui'
 
+export interface RequestWithDetailsBody extends Request {
+  body: {
+    outcome: {
+      detail: string
+      yearCompleted: string
+      yearStarted: string
+      status?: CourseParticipationOutcome['status']
+    }
+    setting: {
+      communityLocation?: CourseParticipationSetting['location']
+      custodyLocation?: CourseParticipationSetting['location']
+      type?: CourseParticipationSetting['type']
+    }
+    source: string
+  }
+}
+
 export default class CourseParticipationUtils {
+  static processDetailsFormData(request: RequestWithDetailsBody): {
```

Comment:
This can also come in a separate PR if preferred, but it would be good to rename the `processedCourseFormData` to `processCourseFormData` to bring it in line

Reply:
Fixed in #250